code-name-taran/src/com/spaghettisoft/component/chess/gamelogic/GameLogic.java

```diff
@@ -0,0 +1,906 @@
+import com.spaghettisoft.component.chess.board.ChessBoard;
+import com.spaghettisoft.component.chess.exceptions.EmtyFieldException;
+import com.spaghettisoft.component.chess.exceptions.IllegalCastingMoveException;
+import com.spaghettisoft.component.chess.exceptions.IllegalEnPassantMoveException;
+import com.spaghettisoft.component.chess.exceptions.IllegalMoveCheckException;
+import com.spaghettisoft.component.chess.exceptions.IllegalMoveException;
+import com.spaghettisoft.component.chess.exceptions.IlligalCapturedKingException;
+import com.spaghettisoft.component.chess.exceptions.IncorectColorMoveException;
+import com.spaghettisoft.component.chess.exceptions.IncorectColorCaptureException;
+import com.spaghettisoft.component.chess.exceptions.KingCheckException;
+import com.spaghettisoft.component.chess.exceptions.SameFieldException;
+import com.spaghettisoft.component.chess.figures.AbstractFigure;
+import com.spaghettisoft.component.chess.figures.AbstractFigure.Color;
+import com.spaghettisoft.component.chess.figures.Bishop;
+import com.spaghettisoft.component.chess.figures.King;
+import com.spaghettisoft.component.chess.figures.Knight;
+import com.spaghettisoft.component.chess.figures.Pawn;
+import com.spaghettisoft.component.chess.figures.Queen;
+import com.spaghettisoft.component.chess.figures.Rook;
+import com.spaghettisoft.component.chess.players.Move;
+import com.spaghettisoft.component.chess.players.Player;
+import com.spaghettisoft.component.chess.userinput.UserInputValidatorAndConvertor;
+import com.spaghettisoft.globals.StaticObjects;
+
+
+public class GameLogic {
+	
+	
+	private static final int SEVEN = 7;
+	private static final int FIFTY_MOVES_RULE_VALUE= 50;
+	private static final int ZERO = 0;
+	private static final int ONE = 1;
+	private static final int MINUS_ONE = -1;
+	private static final int TWO = 2;
+	private static final int THREE = 3;
+	private static final int FOUR = 4;
+												//R		//L 	 //D	 //U      //DR 	  //DL	   //LU		 //RU
+	private static final int[][] directions = { {0, 1}, {0, -1}, {1, 0}, {-1, 0}, {1, 1}, {1, -1}, {-1, -1}, {-1, 1} };
+	
+	private Player whitePlayer;
+	private Player blackPlayer;
+	private ChessBoard chessBoard;
+	private Move lastEnemyMove;
+	private int numberOfFicgureOnChessBoard = 32;
+	private int numberOfMovesWithoutMovePawn = 0;
+	private int numberOfMovesWithoutCapture = 0;
+	private boolean isCurentPlayerInCheck = false;
+
+
+	public GameLogic(Player whitePlayer, Player blackPlayer, ChessBoard chessBoard) {
+		this.whitePlayer = whitePlayer;
+		this.blackPlayer = blackPlayer;
+		this.chessBoard = chessBoard;
+	}
+	
+	
+	public boolean move(Player player, Move move) throws EmtyFieldException, IncorectColorMoveException, IncorectColorCaptureException, IllegalMoveException, KingCheckException, IlligalCapturedKingException, IllegalMoveCheckException, IllegalEnPassantMoveException, SameFieldException, InvalidNameException, IllegalCastingMoveException {

```

Comment:
You can leave the return type of the method void, because you are using exceptions to indicate that a move is invalid.